### ISSUE TITLE
TrainingDataExporter

### ISSUE THREAD

**Issue by wochinge** [RasaHQ/rasa#4584]
I propose a component `TrainingDataExporter` which mirrors the `TrainingDataImporter`.

```python

class TrainingDataExporter:

    @staticmethod
    def save_domain(domain: Domain, path: Text) -> None:
        pass

    @staticmethod
    def save_training_data(training_data: TrainingData, path: Text) -> None:
        pass

    @staticmethod
    def save_stories(stories: List[Story], path: Text) -> None:
        pass

    @staticmethod
    def save_config(config: Dict, path: Text) -> None:
        pass

```

Goals:
- have the option to change the export behavior globally
- users should be able to implement their own module for exporting data
- we should move out the persistence stuff out of our domain objects (improves encapsulation)

Implementations:
- one which exports markdown / yaml
- one which overwrites the last one and exports NLU as json

**Comment by imLew** [comment]
How would this be used? Since the data is already very accessible I imagine this would mainly be a tool that converts between different formats. I definitely like the idea of allowing users to have their own format!

Why put it in the `config.yml` though? My understanding is that that is config for the actual bot. If we have more tools like this that are project specific but not bot specific we should consider having a `project-config` somewhere.

**Comment by wochinge** [comment]
> Why put it in the config.yml though? 

Good point. Currently that's slowly diverging from a bot-config to a general config (DataImporter is also using that). I'm personally not a huge fan of having even more files in a rasa project than we have now, but I see your point.

> How would this be used? Since the data is already very accessible I imagine this would mainly be a tool that converts between different formats. I definitely like the idea of allowing users to have their own format!

That's what I wrote in the `goals`. 
- converting between the data
- making our entity objects / modules leaner

I imagine to configure that in a config and when I use rasa or even rasa x then it would dump data in the specified format.

**Comment by wochinge** [comment]
re-opening this after the short discussion in https://github.com/RasaHQ/rasa/pull/6323
